python/ray/_raylet.pyx

```diff
@@ -452,14 +455,22 @@ cdef execute_task(
                 actor_title = "{}({}, {})".format(
                     class_name, repr(args), repr(kwargs))
                 core_worker.set_actor_title(actor_title.encode("utf-8"))
+            # Ensure no previous signals are still around
+            check_signals()
             # Execute the task.
             with ray.worker._changeproctitle(title, next_title):
                 with core_worker.profile_event(b"task:execute"):
                     task_exception = True
-                    outputs = function_executor(*args, **kwargs)
-                    task_exception = False
+                    try:
+                        outputs = function_executor(*args, **kwargs)
+                        task_exception = False
+                    except KeyboardInterrupt as e:
```

Comment:
Yes, I believe this can be raised on any line of python code

Reply:
It can only be raised in regular python code. When in compiled Cython or C++ code, the interrupts can be observed with `PyErr_CheckSignals` (which `check_signals` does).  As long as I call `check_signals` before python code (the next place it is called is in `store_task_output`, any lingering interrupts will be cleared.